src/Router.sol

```diff
@@ -232,6 +309,62 @@ contract Router is IRouter, EIP712, FeeGenerator {
         return address(agent);
     }
 
+    /// @notice Allow another address to execute on user's behalf
+    /// @param delegatee The address to be delegated
+    /// @param expiry The expiry of the delegation
+    function allow(address delegatee, uint128 expiry) external {
+        IParam.PackedDelegation storage delegation = delegations[msg.sender][delegatee];
+        delegation.updateExpiry(expiry);
+        emit Delegated(msg.sender, delegatee, expiry);
+    }
```

Comment:
```suggestion
    function allow(address delegatee, uint128 expiry) public {
        delegations[msg.sender][delegatee].expiry = expiry;
        emit Delegated(msg.sender, delegatee, expiry);
    }
    
    function disallow(address delegatee) external {
        allow(delegatee, 0);
    }
```
Given the simplicity of the `updateExpiry` logic compared to Permit2's, prefer to have a consistent update method like `[delegatee].nonce = newNonce` as seen on line #364. Additionally, the disallow function can leverage the allow function.